File: Trimorphic Number.py
# solution submitted by Manish Prasad
n = input("Inter the Number ")
n = int(n)
cube = int (n*n*n)

isTrue = True

while(n > 0):
	if n%10 != cube%10:
		isTrue = False
	n = int(n/10)
	cube = int(cube/10)

if isTrue :
	print("True")

else:
	print("False")

File: Trimorphic Number.cpp
#include <iostream> 
using namespace std; 
  bool Fun(long long int n) 
{ 
    long long int cube = n * n * n; 
  
    while (n > 0)  
    { 
        if (n % 10 != cube % 10) 
            return false; 
  
        n = n/10; 
        cube = cube/10; 
    } 
  
    return true; 
} 
int main() 
{ 
    long long int n;
    cin>>n;

    if(Fun(n))
        cout<<"True"<<endl;
    else
        cout<<"False"<<endl;
  
    return 0; 
} 
